.github/workflows/reusable-windows-msi.yml

```
@@ -17,8 +17,11 @@ jobs:
     runs-on: windows-latest
     timeout-minutes: 60
     env:
+      ARCH: ${{ inputs.arch }}
       IncludeFreethreaded: true
     steps:
     - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
     - name: Build CPython installer
-      run: .\Tools\msi\build.bat --doc -${{ inputs.arch }}
+      run: .\Tools\msi\build.bat --doc -"${ARCH}"
```

this probably should also have been

```suggestion
      run: .\Tools\msi\build.bat --doc -"${ARCH}"
      shell: bash
```

maybe this job doesn't run on PRs?

---

Hmm, it ran here:

https://github.com/python/cpython/actions/runs/12255117757/job/34194444625

---

hmmm... are we absolutely confident that the environment variable was properly expanded as part of the command that successfully ran in that workflow?

---

Let's double check. Just heading out to Helsinki Python, I can check tomorrow if no-one beats me to it!

---

Confirmed it's needed.

[Without](https://github.com/hugovk/cpython/actions/runs/12275902522/job/34251971548):

```
Run echo 123 "${ARCH}" 456 && .\Tools\msi\build.bat --doc -"${ARCH}"
123

456
```

[With](https://github.com/hugovk/cpython/actions/runs/12275967901/job/34252185731):

```
Run echo 123 "${ARCH}" 456 && .\Tools\msi\build.bat --doc -"${ARCH}"
123 arm64 456
```

Please see PR https://github.com/python/cpython/pull/127822.